pkg/handler/repository_rpms_test.go

```diff
@@ -0,0 +1,330 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/content-services/content-sources-backend/pkg/api"
+	"github.com/content-services/content-sources-backend/pkg/config"
+	"github.com/content-services/content-sources-backend/pkg/db"
+	test_handler "github.com/content-services/content-sources-backend/pkg/test/handler"
+	mock_dao "github.com/content-services/content-sources-backend/pkg/test/mocks"
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	"github.com/openlyinc/pointy"
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"gorm.io/gorm"
+)
+
+func serveRpmsRouter(req *http.Request, mockDao *mock_dao.RpmDao) (int, []byte, error) {
+	var (
+		err error
+	)
+
+	router := echo.New()
+	router.Use(middleware.RequestIDWithConfig(middleware.RequestIDConfig{
+		TargetHeader: "x-rh-insights-request-id",
+	}))
+	router.Use(config.WrapMiddlewareWithSkipper(identity.EnforceIdentity, config.SkipLiveness))
+	pathPrefix := router.Group(fullRootPath())
+
+	router.HTTPErrorHandler = config.CustomHTTPErrorHandler
+
+	rh := RepositoryRpmHandler{
+		Dao: mockDao,
+	}
+	RegisterRepositoryRpmRoutes(pathPrefix, &rh.Dao)
+
+	rr := httptest.NewRecorder()
+	router.ServeHTTP(rr, req)
+
+	response := rr.Result()
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	return response.StatusCode, body, err
+}
+
+type RpmSuite struct {
+	suite.Suite
+	savedDB *gorm.DB
```

Comment:
I'm pretty sure this database stuff is unused, because we're mocking the dao layer, and can be removed. I see it's also in `repositories_test.go`, but that was added before we started mocking the dao layer.

Reply:
Nice catch! I didn't see at all! updating! :+1: